spec/02-integration/05-proxy/14-upstream-status-header_spec.lua

```diff
@@ -0,0 +1,142 @@
+local helpers = require "spec.helpers"
+local constants = require "kong.constants"
+
+describe(constants.HEADERS.UPSTREAM_STATUS .. " header", function()
+  local client
+
+  setup(function()
+    local bp = helpers.get_db_utils()
+    assert(helpers.dao:run_migrations())
+
+    local service = bp.services:insert {
+      host = helpers.mock_upstream_host,
+      port = helpers.mock_upstream_port,
+      protocol = helpers.mock_upstream_protocol,
+    }
+    assert(service)
+
+    local route1 = bp.routes:insert {
+      protocols = { "http" },
+      service = service,
+      paths = { "/foo" },
```

Comment:
(Minor) one piece of advice here: these tests tend to grow quite fast and using paths such as `/foo` etc quickly makes them unmanageable (several hundreds of lines down in the same files, you might not remember what the `/foo` path maps too).

I would advise for the use of `/status/200` or `/status/500` (or any other more explicit name you may think of) instead. Not really a blocker, but if you care enough to change them, it'd be appreciated.

Reply:
I see your point and have personally experienced this problem in a couple of test files myself. I'll update this to use more descriptive names.